test/actions/katello/host/update_system_purpose_test.rb

```diff
@@ -0,0 +1,38 @@
+require 'katello_test_helper'
+
+module Katello::Host
+  class UpdateSystemPurposeTest < ActiveSupport::TestCase
+    include Dynflow::Testing
+    include Support::Actions::Fixtures
+    include FactoryBot::Syntax::Methods
+    let(:org) { taxonomies(:empty_organization) }
+    let(:library) { katello_environments(:library) }
+    let(:dev) { katello_environments(:dev) }
+    let(:view)  { katello_content_views(:library_dev_view) }
+
+    before :all do
+      User.current = users(:admin)
+      @host = FactoryBot.create(:host, :with_content, :with_subscription, :content_view => view,
+                                :lifecycle_environment => library, :organization => org)
+    end
+
+    describe 'update system purpose' do
+      let(:action_class) { ::Actions::Katello::Host::UpdateSystemPurpose }
+
+      it 'plans' do
+        service_level = 'Standard'
+        purpose_role = 'Red Hat Enterprise Linux Server'
+        purpose_usage = 'Production'
+        action = create_action action_class
+
+        @host.expects(:update_candlepin_associations)
+
+        assert @host.subscription_facet.service_level = 'Standard'
```

Comment:
This isn't exactly what you're looking for. This assertion says 'assert that I can set the service_level to 'Standard', and it's not actually testing your action. What you want to do is, after `plan_action`, assert that the service_level is equal to what was given to the action

Reply:
Updated the test to assert correctly after the plan action and added a 2nd test for the duplicate purpose addon we talked about